File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/container/metrics/Counter.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.vespa.hosted.node.admin.container.metrics;

/**
 * @author freva
 */
public class Counter implements MetricValue {
    private final Object lock = new Object();

    private long value = 0;

    public void increment() {
        add(1L);
    }

    public void add(long n) {
        synchronized (lock) {
            value += n;
        }
    }

    @Override
    public Number getValue() {
        synchronized (lock) {
            return value;
        }
    }
}

File: vespalib/src/vespa/vespalib/metrics/counter.cpp
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
#include "counter.h"
#include "metrics_manager.h"

namespace vespalib {
namespace metrics {


void
Counter::add(size_t count, Point point) const
{
    if (_manager) {
        _manager->add(Increment(std::make_pair(_id, point), count));
    }
}

} // namespace vespalib::metrics
} // namespace vespalib
